compiler-core/src/error.rs

```diff
@@ -394,6 +400,46 @@ numbers and underscores.",
                 write_project(buf, diagnostic);
             }
 
+            Error::InvalidLicense { license, reason } => {
+                let msg;
```

Comment:
Alright, I'll be the first to admit that `msg` is a hack. Essentially on lines 423 and 427, we create new `String`s. These are scoped only to their little block in the match arm, which means that they promptly go out of scope before having an opportunity to do anything useful. What I needed was for these values to live long enough to be able to be passed to `format!`, hence `msg`.

If there's a better way to do this (I'll be there is!), please tell me.

Reply:
You're right, I think I was trying too hard to be performance-sensitive in a place where it doesn't really matter.

So far I've opted for `+` (really just an alias for `push_str`) in the match arms, but if people want a `format!` in there, I'm happy to change it :grin: Heck, this whole function could be done with a `fmt::Formatter`, which I considered.

Reply:
Oh, cool. I didn't realise `+` existed for strings. I've been format!-ing everywhere :) (And some `push_str`) Good to know!